src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp

```diff
@@ -0,0 +1,601 @@
+    {
+        OnDestinationTargetSelect += SpellDestinationTargetSelectFn(spell_aqusirr_erupting_waters_split::SetDest, EFFECT_0, TARGET_DEST_DEST);
+    }
+};
+
+// 264560 - Choking Brine
+class spell_aqusirr_choking_brine : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_CHOKING_BRINE_MISSILE, SPELL_CHOKING_BRINE_MISSILE_2 });
+    }
+
+    void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (GetTargetApplication()->GetRemoveMode() != AURA_REMOVE_BY_ENEMY_SPELL)
+            return;
+
+        Unit* target = GetTarget();
+        target->CastSpell(target, SPELL_CHOKING_BRINE_MISSILE, TRIGGERED_IGNORE_CAST_IN_PROGRESS);
+
+        for (uint8 i = 0; i < 5; ++i)
+            target->CastSpell(target, SPELL_CHOKING_BRINE_MISSILE_2, TRIGGERED_IGNORE_CAST_IN_PROGRESS);
+    }
+
+    void Register() override
+    {
+        AfterEffectRemove += AuraEffectRemoveFn(spell_aqusirr_choking_brine::OnRemove, EFFECT_0, SPELL_AURA_PERIODIC_DAMAGE, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+// 274260 - Surging Rush
+class spell_aqusirr_surging_rush : public SpellScript
+{
+    void HandleScript(SpellEffIndex /*effIndex*/)
+    {
+        Creature* creatureTarget = GetHitCreature();
+        if (!creatureTarget)
+            return;
+
+        creatureTarget->SetDisableGravity(true);
+        creatureTarget->AddUnitState(UNIT_STATE_ROOT);
+        creatureTarget->PlayOneShotAnimKitId(ANIM_KIT_SURGING_RUSH);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_aqusirr_surging_rush::HandleScript, EFFECT_0, SPELL_EFFECT_TRIGGER_ACTION_SET);
+    }
+};
+
+// 264102 - Surging Rush
+class spell_aqusirr_surging_rush_selector : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SURGING_RUSH });
+    }
+
+    void HandleScript(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetHitUnit(), SPELL_SURGING_RUSH, true);
+    }
+
+    void HandleActionSet(SpellEffIndex /*effIndex*/)
+    {
+        Creature* creatureTarget = GetHitCreature();
+        if (!creatureTarget)
+            return;
+
+        creatureTarget->SetDisableGravity(false);
+        creatureTarget->ClearUnitState(UNIT_STATE_ROOT);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_aqusirr_surging_rush_selector::HandleScript, EFFECT_0, SPELL_EFFECT_DUMMY);
+        OnEffectHitTarget += SpellEffectFn(spell_aqusirr_surging_rush_selector::HandleActionSet, EFFECT_2, SPELL_EFFECT_TRIGGER_ACTION_SET);
+    }
+};
+
+// 264941 - Erupting Waters
+class spell_aqusirr_erupting_waters : public SpellScript
+{
+    void HandleDamage(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        caster->SetHealth(caster->GetHealth() - caster->CountPctFromMaxHealth(GetEffectValue()));
```

Comment:
are you sure there is no spell for dealing the damage?

Reply:
same stuff as lady waycrest